PYME/recipes/recipeLayout.py

```diff
@@ -194,7 +194,7 @@ def _descend(node, dg, rdg):
             for child in sorted(children):
                 _descend(child, dg, rdg)
         else:
-            for child in sorted(children):
+            for child in sorted(children, key=lambda x: x if isinstance(x,str) else x.__class__.__name__) :
```

Comment:
would it be reasonable/possible to replace with 
```suggestion
            for child in sorted(children, key=lambda x: str(x)) :
```
(for a class, that will give you `__class__.__name__` and a memory address, which will ensure that sorting is reproducible on a given set of objects)

Using things like `__class__.__name__` explicitly always makes me nervous - it's generally a sign that here be dragons.